### ISSUE TITLE
[Google Search Consol & 애널리틱스 연동]

### ISSUE THREAD

**Issue by seongminlab** [morethanmin/morethan-log#203]
블로그 클론해서 잘 사용하고 있습니다.

빌드 가이드에 참고할 내용이 부족하여 다른 블로그 구현한 코드나 react 코드 간략하게나마 읽어가면서 공부중인데

google search consol 과 analystics 연동 방법을 찾을 수 없어서 요청드립니다.

아래 내용을 보고 따라해봤는데, 연동이 안되는것 같아서요

Deploy on Vercel, with the following environment variables.

NOTION_PAGE_ID (Required): The Notion page Id got from the Share to Web URL.
NEXT_PUBLIC_GOOGLE_MEASUREMENT_ID : For Google analytics Plugin.
NEXT_PUBLIC_GOOGLE_SITE_VERIFICATION : For Google search console Plugin.

구글 애널리틱스에 등록하여,  Measurment_ID   를  G-XXXXX 를 받았는데 
Vercel 환경변수에   
NEXT_PUBLIC_GOOGLE_MEASUREMENT_ID     value:  G-XXXXX

구글 서치 콘솔에서는 도메인 등록하여  TXT 레코드를   google-site-verification=KOXXXXX  를 받았고
NEXT_PUBLIC_GOOGLE_SITE_VERIFICATION    value: KOXXXXX

이렇게 등록해 둔 상황입니다.  하루 지났는데 아직 인증이 안된것 같은데,  혹시 이렇게 진행 하는 것이 맞을까요??

vercel 에서  구글 서치 콘솔 사용 방법을 찾아봐도 찾지 못해서 가이드 부탁드립니다.

감사합니다

**Comment by seongminlab** [comment]
site.config.js 파일에서 google analystics : true로 변경해 활성화 되는것 확인했습니다.


여전히 google search consol은 연결이 안되는데,  환경변수 입력이 잘못된건지, 혹은 다른 설치 방식 (TXT 레코드 말고 다른 방법이 적합한지 등 ) 이 있는건지 알려주시면 감사하겠습니다

**Comment by flavono123** [comment]
@seongminlab 
![image](https://github.com/morethanmin/morethan-log/assets/20619567/97c5c321-2ee8-4011-a41f-8347aaa2c7bb)
site.config.js 에서 둘 다 활성화 하셨나요?

**Comment by seongminlab** [comment]
@flavono123    

네 둘다 활성화 했습니다.  google search console 에서 HTML 테그로 인증하는걸 잘못해서,  해당 verification이 안된것 같습니다.  도메인 TXT 레코드로는 해결이 안되더라구요

이 부분에 대해서는 구체적으로 문서가 필요해보이긴 합니다. vercel을 통한 도메인 사용이다보니, 소유권 인증 부분을 잘 모르면 사용이 어렵네요 


sitemap.xml 파일 제출했고

public/robots.txt 파일도 수정했습니다.
public/sitemap.xml 은 제 주소 링크가 아닌데, 내용을 변경해야될지 모르겠네요

search console 에 제출한 sitemap.xml 은 아직 상태가 가져올수 없음 상태고
다행히 등록 이후에 /sitemap.xml 은 생성된것 같아 보입니다.
https://minari-blog.vercel.app/sitemap.xml

sitemap을 통한 크롤링 연계와 별개로 HTML에  블로그 검색 가능하도록 메타테그를 달고 싶은데, 방법이 있을지 모르겠습니다.

**Comment by flavono123** [comment]
@seongminlab 
> google search console 에서 HTML 테그로 인증하는걸 잘못해서,
> HTML에 블로그 검색 가능하도록 메타테그를 달고 싶은데,


저는 아마 위 방법으로 한거 같구요.
일단 sitemap 제출 성공까지 했는데요.
참고하시라고 제가 정리한거 올려 드려요(이걸 노션 morethan-log 블로그 글로 썼는데 노출이 안돼서 공유를 못하네요 😑 )

---

![image](https://github.com/morethanmin/morethan-log/assets/20619567/cd35f984-da84-4961-87d4-fef5900f3707)
> 구글 서치콘솔은 속성 추가 시 “URL 접두어”를 선택하고 마찬가지로 Vercel 블로그 주소를 써준다(나의 Vercel 주소는 이미 속성 등록되어 있어서, 캡쳐에선 과거 블로그 주소를 사용함). 그리고 소유권 확인에서 다른 확인 방법 중 첫번째인 “HTML 태그”를 펼치고 meta 태그 안의 content 어트리뷰트의 값을 복사한다. 꽤 긴 해시 값이다. 이걸 NEXT_PUBLIC_GOOGLE_SITE_VERIFICATION 에 써준다.

---

> 이 부분에 대해서는 구체적으로 문서가 필요해보이긴 합니다.

천만퍼센트 동의합니다.
related issues: 
- #222
- #258

**Comment by seongminlab** [comment]
@flavono123  네 해당 방법으로 저도 했고

메타태그 이야기는 저거 말고, sitemap 을 통한 색인 생성 방식이 아니라 
블로그 html 문서 상단에 크롤링시에 접근 가능하도록 별도 테그를 달아두고 싶다는 이야기였습니다.
아마 tx 파일을 직접 수정하거나 해야할것 같은데
아직 SEO에 대한 이해가 부족해서  색인을 별도로 다는 방법에 대한 문서가 구글에서 찾아봐도 잘 설명이 안되어있어서 어렵네요
vercel쪽 문서를 읽어봐야 할것 같습니다.

**Comment by flavono123** [comment]
> 블로그 html 문서 상단에 크롤링시에 접근 가능하도록 별도 테그를 달아두고 싶다는 이야기였습니다.

이건 feature request 하거나 PR을 만들어야겠군요.

> 아직 SEO에 대한 이해가 부족해서 색인을 별도로 다는 방법에 대한 문서가 구글에서 찾아봐도 잘 설명이 안되어있어서 어렵네요

혹시 저처럼 서치 콘솔에 사이트맵 제출 성공했지만 검색 결과에 안뜬다는 말씀이신가요?
